### PR TITLE
Codechange: Pass by reference to UpdateWidgetSize.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Parameters are passed to UpdateWidgetSize() by pointer, but they are not optional and are always present.

## Description

Replace pointers with references instead.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by PeterN** [comment]
As mentioned, it's 3 in,out parameters. I've updated the doxygen to note this.

File: src/company_gui.cpp [hunk 1]
```diff
@@ -1840,57 +1840,57 @@ struct CompanyInfrastructureWindow : Window
...
+	void UpdateWidgetSize(WidgetID widget, Dimension &size, [[maybe_unused]] const Dimension &padding, [[maybe_unused]] Dimension &fill, [[maybe_unused]] Dimension &resize) override
 	{
 		const Company *c = Company::Get((CompanyID)this->window_number);
 
 		switch (widget) {
 			case WID_CI_RAIL_DESC: {
 				uint lines = 1; // Starts at 1 because a line is also required for the section title
 
-				size->width = std::max(size->width, GetStringBoundingBox(STR_COMPANY_INFRASTRUCTURE_VIEW_RAIL_SECT).width + padding.width);
+				size.width = std::max(size.width, GetStringBoundingBox(STR_COMPANY_INFRASTRUCTURE_VIEW_RAIL_SECT).width);
```
Comment (rubidium42) [inline src/company_gui.cpp]: Should this padding be removed? There are many cases of this in this function.

File: src/group_gui.cpp [hunk 2]
```diff
@@ -400,56 +400,56 @@ class VehicleGroupWindow : public BaseVehicleListWindow {
...
 				this->ComputeGroupInfoSize();
-				resize->height = GetVehicleListHeight(this->vli.vtype, this->tiny_step_height);
-				size->height = 4 * resize->height;
+				resize.height = GetVehicleListHeight(this->vli.vtype, this->tiny_step_height);
+				size.height = 4 * resize.height;
 				break;
 
 			case WID_GL_GROUP_BY_DROPDOWN:
-				size->width = GetStringListWidth(this->vehicle_group_by_names) + padding.width;
+				size.width = GetStringListWidth(this->vehicle_group_by_names) + padding.width;
```
Comment (rubidium42) [inline src/group_gui.cpp:433]: In other places `padding.width` gets removed. Why is it kept here?

File: src/news_gui.cpp [hunk 3]
```diff
@@ -1099,22 +1099,28 @@ struct MessageHistoryWindow : Window {
...
+			size.width = std::max(200u, size.width); // At least 200 pixels wide.
 		}
 	}
 
+	void OnPaint() override
+	{
+		this->OnInvalidateData(0);
+		this->DrawWidgets();
+	}
+
```
Comment (rubidium42) [inline src/news_gui.cpp]: What has this to do with the scope of this patch?

**Review by rubidium42 — COMMENTED** [review]
I'm also seeing changes to padding happening in several places, but not everywhere. So what's up with that? Should all the padding stay, or should all the padding go? The two places I put the comments to are definitely not the only places where that happened.

File: src/company_gui.cpp [hunk 4]
```diff
@@ -1840,57 +1840,57 @@ struct CompanyInfrastructureWindow : Window
...
+	void UpdateWidgetSize(WidgetID widget, Dimension &size, [[maybe_unused]] const Dimension &padding, [[maybe_unused]] Dimension &fill, [[maybe_unused]] Dimension &resize) override
 	{
 		const Company *c = Company::Get((CompanyID)this->window_number);
 
 		switch (widget) {
 			case WID_CI_RAIL_DESC: {
 				uint lines = 1; // Starts at 1 because a line is also required for the section title
 
-				size->width = std::max(size->width, GetStringBoundingBox(STR_COMPANY_INFRASTRUCTURE_VIEW_RAIL_SECT).width + padding.width);
+				size.width = std::max(size.width, GetStringBoundingBox(STR_COMPANY_INFRASTRUCTURE_VIEW_RAIL_SECT).width) + padding.width;
```
Comment (rubidium42) [inline src/company_gui.cpp]: Remaining padding change.

File: src/newgrf_gui.cpp [hunk 5]
```diff
@@ -2231,8 +2231,8 @@ struct ScanProgressWindow : public Window {
 				SetDParamMaxDigits(1, 4);
 				/* We really don't know the width. We could determine it by scanning the NewGRFs,
 				 * but this is the status window for scanning them... */
-				size->width = std::max<uint>(size->width, GetStringBoundingBox(STR_NEWGRF_SCAN_STATUS).width + padding.width);
```
Comment (rubidium42) [inline src/newgrf_gui.cpp:2234]: Padding change.

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**